control_protocol.md

```diff
@@ -280,28 +280,55 @@ For the format of the Messages, see {ref}`message-layer`.
 
 ##### Coordinator sign in
 
-:::{note}
-TODO
-:::
+A Coordinator `Co1`joining a network follows a few steps:
+1. It signs in to one Coordinator `Co2` of the Network.
+2. It sends a CO_TELL_ALL message to `Co2`, to tell all other Coordinators about `Co1`s address.
+3. `Co2` tells all the Coordinators signed in (`Co3`, `Co4`...) about `Co1` with a CO_NEW message.
+4. These other Coordinators (`Co3`, `Co4`...) sign in to `Co1`.
+5. All Coordinators are connected to all others.
```

Comment:
Why not
```suggestion
A Coordinator `Co1`joining a network follows a few steps:
1. It signs in to one Coordinator `Co2` of the Network.
2. If successful, `Co2` sends a list of all Coordinators (and their addresses) that it knows
3. `Co1` signs in to all Coordinators on this list that it does not know yet (step 1)
4. All Coordinators are connected to all others.
```
* Same outcome
* The responsibility for connecting remains with the connecting Coordinator
* There is not arbitrary difference between `Co2` and `Co3+` w.r.t. to the connection topology, they all were connected by `Co1`'s actions.

Reply:
That requires, that every Coordinator stores the addresses of the Coordinators.
In my setup, all Coordinators know their own address and can forget the other addresses after connecting.

> No "tell-all" message type needed. 

Yes, but you need a "give me all Coordinators" message type. 

Oh, or it is part of the local directory...?

Reply:
> That requires, that every Coordinator stores the addresses of the Coordinators.
In my setup, all Coordinators know their own address and can forget the other addresses after connecting.

Indeed... hmmm..

> Yes, but you need a "give me all Coordinators" message type.

No, this is just part of the sign-in flow for a coordinator.

> Oh, or it is part of the local directory...?

We would need a separate mapping from Coordinator full name to Coordinator (router) address.

Reply:
Here another suggestion

```suggestion
A Coordinator `Co1`joining a network follows a few steps:
1. `Co1` signs in to one Coordinator `Co2` of the Network.
2. After successful a sign in handshake, `Co1` requests a list of all other Coordinators known to `Co2` (`Co3`, `Co4`, ...). 
3. `Co1` signs in to all the Coordinators on the list of `Co2`
4. `Co1` requests the lists of the connected Coordinators from all now connected Coordinators (except from `Co2`, it already has this one)
5. `Co1` compares the lists, and notifies Coordinators which have an incomplete set of the missing Coordinators
6. Coordinators with previously incomplete sets do sign ins to the new Coordinators they have been told about, until all Coordinators are connected to all others
```
In this case, if some Coordinator somehow drops out, those lists come out of sync, at every new sign in of a new Coordinator, the whole system brings itself back into sync. 
The only new message type would be "hey, you are missing a few more connections". 
For clarity, I separated the request for the list of connected Coordinators from the initial `SIGNIN`, although that could be part of the `SIGNIN` flow. 

> We would need a separate mapping from Coordinator full name to Coordinator (router) address.

Remember that we have a separate DEALER for every other Coordinator's connected ROUTER socket, yes, we need (in the implementation) to keep track of which DEALER socket belongs to which remote Coordinator's ROUTER address, but we need to do that anyways.

Reply:
Let us discuss in #44

Reply:
Sounds good, let us just drop a TODO annotation here and move on, so that we don't delay the PR more.